### PR TITLE
[1LP][RFR] New Test: Test vm with quota for disk removal

### PR DESCRIPTION
**Pull request by ganeshhubale**
## Purpose or Intent
- Testing quota not fails after removing disk from vm
### PRT Run
 {{ pytest: cfme/tests/infrastructure/test_tenant_quota.py::test_quota_not_fails_after_vm_reconfigure_disk_remove --use-template-cache -qsvvv  --long-running --use-provider=complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 16558b521432b95f2192a16002de9bc3f1b421a8

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 1cc2101400f03dd956c481ddd3d3efcc58e890f4

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 82f4a330ada283ab5480fd705b67f9c1cb2940f9

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit bddf837c2847e3e5c7d04f6f91470e5f5370e8d5

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 732e0727c0f31706c1e3a30a3657e3a0a31c346e

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 741659708f6fb7d5a08db16e7f3b4421dddcaa18

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 2dc808044afbdfb62486454500c50015eddd14eb

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 918565768e3a7751e912bb4f8a066e2931453ff6

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 306904d2482c77e96f85c4d537b2ca865b74b9ad

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0b25a6b0ddf5a8a9f70836519e88a1e0d1bc4f7a

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 368bfc68b694caa6b05d22c67a2d8f561ff0f97f

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by valaparthvi** [comment]
Code changes looks good, but we need to restart the PRT, since the tests were uncollected, I also made change to PRT test name because the provider you're using is `VMWareProvider` but you were using RHEV provider in `--use-provider` flag.

**Comment by valaparthvi** [comment]
I am moving this to WIPTEST, looks like there some errors that need to be fixed.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit e2389f4204140de01b761e32258e349c661f0898

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 21e02a8f66bcc3be2d3d939570369bcc81d91272

The *local* fixture **`vm_state`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

The *local* fixture **`full_vm`** is used in the following files:
- **cfme/tests/infrastructure/test_tenant_quota.py**
    - *vm_state*
    - *test_quota_not_fails_after_vm_reconfigure_disk_remove*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 1]
```diff
@@ -506,3 +507,85 @@ def test_quota_exceed_mail_with_more_info_link(configure_mail, appliance, provid
         provision_request = appliance.collections.requests.instantiate(request_description)
         provision_request.wait_for_request(method='ui')
         assert provision_request.row.reason.text == "Quota Exceeded"
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1644351])
+@pytest.mark.provider([VMwareProvider], override=True, selector=ONE)
```
Comment (valaparthvi) [inline cfme/tests/infrastructure/test_tenant_quota.py]: There is no need to use `override=True` anymore.

File: cfme/tests/infrastructure/test_tenant_quota.py [hunk 2]
```diff
@@ -506,3 +507,85 @@ def test_quota_exceed_mail_with_more_info_link(configure_mail, appliance, provid
...
+        # Remove disk request verification
+        wait_for(
+            remove_disk_request.is_succeeded,
+            timeout=360,
+            delay=45,
+            message="confirm that previously-added disk was removed",
+        )
+
+        # Remove disk UI verification
+        wait_for(
```
Comment (valaparthvi) [inline cfme/tests/infrastructure/test_tenant_quota.py]: Do you need to put a wait_for here even after the disk reconfig request was succesfully completed? Can you not just assert the condition?

Comment (ganeshhubale) [reply cfme/tests/infrastructure/test_tenant_quota.py]: Need to wait to check if that disk is removed or not.

**Pull request merged**